File: src/main/java/com/indiepost/service/CategoryService.java
package com.indiepost.service;

import com.indiepost.dto.CategoryDto;
import com.indiepost.model.Category;

import java.util.List;

/**
 * Created by jake on 8/4/16.
 */
public interface CategoryService {
    void save(Category category);

    void delete(Category category);

    Category getReference(Long id);

    Category findById(Long id);

    Category findBySlug(String slug);

    List<Category> findAll();

    List<Category> findByParent(Category parent);

    List<CategoryDto> getDtoList();
}

File: src/main/java/com/indiepost/service/CategoryService.kt
package com.indiepost.service

import com.indiepost.dto.CategoryDto
import com.indiepost.model.Category

/**
 * Created by jake on 8/4/16.
 */
interface CategoryService {

    fun getDtoList(): List<CategoryDto>

    fun save(category: Category)

    fun delete(category: Category)

    fun getReference(id: Long?): Category?

    fun findById(id: Long?): Category?

    fun findBySlug(slug: String?): Category?

    fun findAll(): List<Category>

    fun findByParent(parent: Category): List<Category>
}